starknet_py/net/schemas/rpc.py

```diff
@@ -444,67 +473,93 @@ class TypesOfTransactionsSchema(OneOfSchema):
     }
 
 
-class PendingStarknetBlockSchema(Schema):
-    parent_block_hash = Felt(data_key="parent_hash", required=True)
+class PendingBlockHeaderSchema(Schema):
+    parent_hash = Felt(data_key="parent_hash", required=True)
+    timestamp = fields.Integer(data_key="timestamp", required=True)
     sequencer_address = Felt(data_key="sequencer_address", required=True)
-    transactions = fields.List(
-        fields.Nested(TypesOfTransactionsSchema(unknown=EXCLUDE)),
-        data_key="transactions",
-        required=True,
+    l1_gas_price = fields.Nested(
+        ResourcePriceSchema(), data_key="l1_gas_price", required=True
     )
+    l1_data_gas_price = fields.Nested(
+        ResourcePriceSchema(), data_key="l1_data_gas_price", required=True
+    )
+    l1_da_mode = L1DAModeField(data_key="l1_da_mode", required=True)
+    starknet_version = fields.String(data_key="starknet_version", required=True)
+
+
+class BlockHeaderSchema(Schema):
+    block_hash = Felt(data_key="block_hash", required=True)
+    parent_hash = Felt(data_key="parent_hash", required=True)
+    block_number = fields.Integer(data_key="block_number", required=True)
+    new_root = Felt(data_key="new_root", required=True)
     timestamp = fields.Integer(data_key="timestamp", required=True)
+    sequencer_address = Felt(data_key="sequencer_address", required=True)
     l1_gas_price = fields.Nested(
         ResourcePriceSchema(), data_key="l1_gas_price", required=True
     )
+    l1_data_gas_price = fields.Nested(
+        ResourcePriceSchema(), data_key="l1_data_gas_price", required=True
+    )
+    l1_da_mode = L1DAModeField(data_key="l1_da_mode", required=True)
     starknet_version = fields.String(data_key="starknet_version", required=True)
```

Comment:
Nit: Would it be reasonable to add an intermediate class to avoid some code duplication?

Reply:
On the other hand, it's "copied" from starknet-spec. IMO it's cleaner when we keep this basic spec class implement explicit

Reply:
We can consider introducing an additional class, such as `BlockHeaderCommon` in #1231